iceoryx_utils/test/moduletests/test_cxx_stack.cpp

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_utils/cxx/stack.hpp"
+#include "test.hpp"
+using namespace iox;
+using namespace ::testing;
+
+namespace
+{
+class TestClass
+{
+  public:
+    TestClass() noexcept = default;
+    TestClass(int a, int b, int c) noexcept
+        : m_a(a)
+        , m_b(b)
+        , m_c(c)
+    {
+    }
+
+    bool operator==(const TestClass& rhs) const noexcept
+    {
+        return m_a == rhs.m_a && m_b == rhs.m_b && m_c == rhs.m_c;
+    }
+    int m_a = 0, m_b = 0, m_c = 0;
+};
+
+class stack_test : public Test
+{
+  public:
+    static constexpr uint64_t STACK_SIZE = 10U;
+    cxx::stack<TestClass, STACK_SIZE> m_sut;
+};
+} // namespace
+
+TEST_F(stack_test, isEmptyOnCreation)
```

Comment:
I'm not sure if it's a rule but most of the tests have names starting with capital letters

Reply:
Mmhh we do not have this rule ... And at the moment I do not see the necessity for this rule since the test name is in either case good readable. 

But sometimes I think if snake case for test names wouldn't be better readable.
What do you think?!

Reply:
yes, snake case might be better